### PR TITLE
Author statistics extension

### PR DESCRIPTION
**Pull request by duncandewhurst**

### CODE REVIEW

**Comment by duncandewhurst** [comment]
Based on the discussion in https://github.com/open-contracting/ocds-extensions/issues/201, I've coped the `Statistic` definition from the bids extension. Based on https://github.com/open-contracting/european-union-support/issues/173#issuecomment-1533415489, I haven't copied the codes from the EU codelist to `statistic.csv`. However, that results in a test error because `statistic.csv` is missing. To resolve that error, I've added the bids extension to `testDependencies`. The tests now fail with the following error:

> unexpectedly overwrites /definitions/Statistic/title

@jpmckinney can you advise on the best approach here?

**Comment by jpmckinney** [comment]
Hmm, if we depend on the bids extension, we can't overwrite its Statistic definition.

We can either:

1. Depend on the bids extension, and don't copy the Statistic definition. Any new statistics go in a `+statistic.csv`. The downside is that a user of this extension necessarily gets a bunch of bid fields. This is maybe acceptable - depending on whether we want to edit any of the bid extension's Statistic fields.
2. Don't depend on the bids extension. Copy the Statistic definition and statistic.csv file. The downside is that if a user of this extension also uses the bids extension, then they will overwrite each other based on their order in the `extensions` array. Also, it becomes harder to keep the schema in sync.

I think 1 is fine.

**Comment by duncandewhurst** [comment]
Option 1 sounds good. The only edit I made was to the description of the `Statistic` definition because the original description uses 'tender' rather than 'contracting process', but I think we can live with that. Otherwise, the fields on `Statistic` aren't specific to bids.

>  Any new statistics go in a `+statistic.csv`

The guidance for BT-636-LotResult maps the codes from the EU's [irregularity type codelist](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/irregularity-type) to `Statistic.measure`, which references the open `statistic.csv` codelist, and maps the code labels to `Statistic.notes`. Do we need to add the codes from the EU codelist to `+statistic.csv`? I think not since at least one code ([nc-proc-eap](https://op.europa.eu/web/eu-vocabularies/concept/-/resource?uri=http://publications.europa.eu/resource/authority/irregularity-type/nc-proc-eap)) is specific to the EU and elsewhere you mentioned that it was OK to use external codes as long as the source was mentioned in the publication policy, but want to check that we're on the same page.

**Comment by duncandewhurst** [comment]
> > Any new statistics go in a `+statistic.csv`
> 
> The guidance for BT-636-LotResult maps the codes from the EU's [irregularity type codelist](https://op.europa.eu/en/web/eu-vocabularies/concept-scheme/-/resource?uri=http://publications.europa.eu/resource/authority/irregularity-type) to `Statistic.measure`, which references the open `statistic.csv` codelist, and maps the code labels to `Statistic.notes`. Do we need to add the codes from the EU codelist to `+statistic.csv`? I think not since at least one code ([nc-proc-eap](https://op.europa.eu/web/eu-vocabularies/concept/-/resource?uri=http://publications.europa.eu/resource/authority/irregularity-type/nc-proc-eap)) is specific to the EU and elsewhere you mentioned that it was OK to use external codes as long as the source was mentioned in the publication policy, but want to check that we're on the same page.

Oh wait, ignore me, I see that you already agreed with the approach of not adding codes to `statistic.csv` in https://github.com/open-contracting/european-union-support/issues/116#issuecomment-1292300479.

**Review by odscjen — APPROVED** [review]
This all looks fine but if we use it as it is we'll need to rewrite some of the guidance.yaml. Currently all the statistics BT's, both both bids and complaints, refer to "#Add a statistic" in `operations.md`. That says to place ALL statistics in the top-level `statistics` array and set the `scope` to either 'bids' or 'complaints' as appropriate. But the way this has been dealt with here means bid related statistics will still sit in `bid.statistics` and we've not added `scope` to that extension.

I'm approving because the extension looks fine, but just needed to note that we'll need to do more work in guidance.yaml as a result of it

**Comment by jpmckinney** [comment]
@odscjen Do you want to create an issue on european-union-support for that additional work?

File: README.md [hunk 1]
```diff
@@ -1,180 +1,29 @@
-# OCDS Extension Template Repository
+# Statistics
 
-[OCDS extensions](https://standard.open-contracting.org/latest/en/extensions/) can be used to declare additional fields, objects and sections that will be included in an OCDS file, but which are not in the core schema, and to add or extend codelists. When any OCDS publication cites an extension in its' package metadata, it should be validated against the schema with that extension applied. 
+Adds a top-level `statistics` field to describe statistics about the contracting process.
```
Comment (jpmckinney) [inline README.md]:
```suggestion
Adds a top-level statistics array to describe statistics about the contracting process.
```

File: extension.json [hunk 2]
```diff
@@ -1,30 +1,28 @@
 {
   "name": {
-    "en": "Name of extension (required in at least one language)"
+    "en": "Statistics"
   },
   "description": {
-    "en": "A brief description of what this extension does, this will be pulled into the main docs as a link (required in at least one language)"
+    "en": "Adds a top-level statistics field to describe statistics about the contracting process."
```
Comment (jpmckinney) [inline extension.json]:
```suggestion
    "en": "Adds a top-level statistics array to describe statistics about the contracting process."
```

File: release-schema.json [hunk 3]
```diff
@@ -1 +1,35 @@
...
+      "items": {
+        "$ref": "#/definitions/Statistic"
+      },
+      "minItems": 1,
+      "uniqueItems": true
+    }
+  },
+  "definitions": {
+    "Statistic": {
+      "properties": {
```
Comment (jpmckinney) [inline release-schema.json:16]: Aren't we missing a bunch of properties here?

Comment (duncandewhurst) [reply release-schema.json:16]: We're depending on the bids extension so this is the only extra property. Did I miss something?

Comment (jpmckinney) [reply release-schema.json:16]: Oh, right!

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**